seqr/views/apis/dataset_api.py

```diff
@@ -136,3 +139,10 @@ def _get_samples_json(updated_samples, inactivated_sample_guids, project_guid):
             ind.guid: {'sampleGuids': [s.guid for s in ind.sample_set.all()]} for ind in individuals
         }
     return response
+
+# Service account access
+
+@service_account_access
+def sa_add_variants_dataset(request, project_guid):
+    return add_variants_dataset_handler_base(request, project_guid)
```

Comment:
I was thinking about this too, the original design in #130 is exclusive, endpoints can only be regular OR service-account access - there was no room for hybrid because it allowed endpoints to potentially support multiple types of authentication - and then might blur the line for who can use endpoints.

Agree that this kind of refactoring feels a little bit gross, but it's the best way I can see to do it within the initial constraints of how service-account access works